### PR TITLE
Add undefined to EntryListener interface

### PR DESCRIPTION
**Pull request by srknzl**
reference: https://github.com/hazelcast/hazelcast-nodejs-client/issues/1286

### CODE REVIEW

File: src/proxy/EntryListener.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ export interface EntryListener<K, V> {
     mapEvicted?: MapEventListener<K, V>;
     mapCleared?: MapEventListener<K, V>;
 
-    [event: string]: EntryEventListener<K, V> | MapEventListener<K, V>;
+    [event: string]: EntryEventListener<K, V> | MapEventListener<K, V> | undefined;
```
Comment (mdumandag) [inline src/proxy/EntryListener.ts:34]: Why do we need this line? I mean, aren't all possible events defined as properties above?

Comment (srknzl) [reply src/proxy/EntryListener.ts:34]: It is defined but they can be undefined. This line is the general definition. It means any key can be mapped to either EntryEventListener, MapEventListener or undefined. A user mentioned if we don't put undefined here, typescript compilation fails for him.

Comment (srknzl) [reply src/proxy/EntryListener.ts:34]: We may not need this line at all. Let me think

Comment (srknzl) [reply src/proxy/EntryListener.ts:34]: We need this line in order to be able to index the type like `entryListener[stringVariable]`

Comment (srknzl) [reply src/proxy/EntryListener.ts:34]: Yes, removing this line may make the code more complex because we will need more switch/case blocks.

**Review by mdumandag — APPROVED** [review]

**Pull request merged**